mmv1/third_party/terraform/services/composer/resource_composer_environment.go.erb

```diff
@@ -338,39 +338,34 @@ func ResourceComposerEnvironment() *schema.Resource {
 										Optional:    true,
 										Computed:    true,
 										ForceNew:    true,
-										ConfigMode:  schema.SchemaConfigModeAttr,
 										MaxItems:    1,
 										Description: `Configuration for controlling how IPs are allocated in the GKE cluster. Cannot be updated.`,
 										Elem: &schema.Resource{
 											Schema: map[string]*schema.Schema{
 												"use_ip_aliases": {
 													Type:         schema.TypeBool,
-													Optional:     true,
+													Optional:     true,													
 													ForceNew:     true,
-													AtLeastOneOf: composerIpAllocationPolicyKeys,
```

Comment:
I believe this is originally added to avoid users from sending empty blocks, so it wouldn't be breaking to remove. However, I'm still a bit confused why this will cause the error you see in https://github.com/hashicorp/terraform-provider-google/issues/15456#issuecomment-2274414644, why it makes all subfields as required.

Is there any potential way that you can check locally if this will fix the issue you're facing in CDKTF  before merging the change to the providers?

Reply:
Yeah, I confirmed this fixes the issue with CDKTF/terraform compliant JSON. I'll be very honest, I'm not entirely certain why this fixes it, because it really seems like this _should_ work as intended, but yeah, this fixes it. 

I don't have any definitive proof, but I have a feeling this is probably due to the lack of a 1:1 translation between terraform compliant JSON and HCL, and something that may be a lower level terraform bug fix. 
https://developer.hashicorp.com/terraform/language/attr-as-blocks#in-json-syntax

Also, while I don't have any definitive proof as to why the GKE ip_allocation_policy resource is structured in a similar way to this PR, I wouldn't be surprised if it was due to this as well. 

https://github.com/hashicorp/terraform-provider-google/blob/main/google/services/container/resource_container_cluster.go#L1443-L1536

Reply:
It looks like 6.1.0 fixes this, it attempts to recreate the cluster but it looks like there was something also done on GCP's backend that removed the ability to use `-`